src/mucjoindlg.cpp

```diff
@@ -56,12 +56,19 @@ MUCJoinDlg::MUCJoinDlg(PsiCon *psi, PsiAccount *pa) : QDialog(nullptr), nickAlre
     ui_.ck_history->hide();
     joinButton_ = ui_.buttonBox->addButton(tr("&Join"), QDialogButtonBox::AcceptRole);
     joinButton_->setDefault(true);
-    timer_ = new QTimer(this);
-    timer_->setInterval(timeout);
-    connect(timer_, &QTimer::timeout, this, [this]() {
-        account_->groupChatLeave(jid_.domain(), jid_.node());
-        error(404, tr("No response from server for %1 seconds").arg(timeout / 1000));
-    });
+    if (MucInterval) {
+        if (!joinTimer_) {
+            joinTimer_ = new QTimer(this);
+            connect(joinTimer_, &QTimer::timeout, this, [this]() {
+                account_->groupChatLeave(jid_.domain(), jid_.node());
+                error(404, tr("No response from server for %1 seconds").arg(MucInterval));
+            });
+        }
+        joinTimer_->setInterval(MucInterval * 1000);
+    } else {
+        delete joinTimer_;
```

Comment:
There is no reason to delete this in constructor if it was never created and likely it will crash since joinTimer_ isn't initialized in advance